### PR TITLE
Add in a method to zero the memory of a slice

### PR DESCRIPTION
**Pull request by borrrden**

### CODE REVIEW

File: API/fleece/slice.hh [hunk 1]
```diff
@@ -238,7 +238,8 @@ namespace fleece {
 
         void setBuf(const void *b NONNULL) noexcept          {pure_slice::setBuf(b);}
         void setSize(size_t s) noexcept                      {pure_slice::setSize(s);}
-        void shorten(size_t s)                      {assert_precondition(s <= size); setSize(s);}
+        void shorten(size_t s)                               {assert_precondition(s <= size); setSize(s);}
+        void clear()                                         { memset((void *)buf, 0, size); }
```
Comment (snej) [inline API/fleece/slice.hh:242]: The name `clear` doesn't seem right to me. STL containers have `clear` methods that reset them to zero size, and LiteCore has some methods like that too. How about `wipe` or `zeroFill`?

File: API/fleece/Fleece.hh [hunk 2]
```diff
@@ -578,6 +578,8 @@ namespace fleece {
         const alloc_slice& data() const                 {return *this;}
         explicit operator bool () const                 {return Dict::operator bool();}
 
+        void clear() { alloc_slice::clear(); }
```
Comment (snej) [inline API/fleece/Fleece.hh:581]: Nit: Make the indentation of the method body consistent with nearby lines

File: API/fleece/slice.hh [hunk 3]
```diff
@@ -350,6 +351,7 @@ namespace fleece {
         void reset(size_t);
         void resize(size_t newSize);
         void append(pure_slice);
+        void clear() { memset((void *)buf, 0, size); }
```
Comment (snej) [inline API/fleece/slice.hh:354]: Nit: Make the indentation of the method body consistent with nearby lines

**Review by snej — CHANGES_REQUESTED** [review]

**Comment by davemilter** [comment]
Sorry for the interference,
but if this is for security, then `memset` or loop should not be used:
 https://rules.sonarsource.com/c/type/Vulnerability/RSPEC-5798

**Comment by borrrden** [comment]
This intrusion is most welcome.  I never would have known about this otherwise....

**Pull request merged**